src/QsCompiler/QirGeneration/Subtransformations/ExpressionKindTransformation.cs

```diff
@@ -386,6 +386,252 @@ IValue CopyAndUpdateUdt(TupleValue originalValue)
             return ResolvedExpressionKind.InvalidExpr;
         }
 
+        /// <summary>
+        /// Creates a string literal that contains the given values as interpolation arguments.
+        /// The given string is expected to consist of text interspaced with integer constants between curly brackets.
+        /// The content between non-escaped curly brackets is parsed as integer,
+        /// and replaced with the string representation of the interpolation argument at that index.
+        /// Registers the built string with the scope manager.
+        /// </summary>
+        internal static IValue CreateStringLiteral(GenerationContext sharedState, string str, params IValue[] exs)
+        {
+            static (int, int, int) FindNextExpression(string s, int start)
+            {
+                while (true)
+                {
+                    var i = s.IndexOf('{', start);
+                    if (i < 0)
+                    {
+                        return (-1, s.Length, -1);
+                    }
+                    else if ((i == start) || (s[i - 1] != '\\'))
+                    {
```

Comment:
Just in case I want to make sure that the code correctly handles the case when `s` contains `"..\\{.."`, i.e. the backslash `\` is escaped, but the brace `{` is not escaped, and should be handled/parsed.

Reply:
Thanks for the catch! I switched it to enter the block if there is an even number of backslashes before the {.